include/nbl/video/utilities/CAssetConverter.h

```diff
@@ -822,6 +826,13 @@ class CAssetConverter : public core::IReferenceCounted
 				return {};
 			}
 
+			// If you absolutely need to avoid some memory type for your image or buffer, you can specify a mask here
+			// one example would be to use HOST_VISIBLE to make sure your buffer can be written directly and doesn't need to go through staging
+			virtual inline uint32_t constrainMemoryTypeBits(const size_t groupCopyID, const asset::IAsset* canonicalAsset, const core::blake3_hash_t& contentHash, const IDeviceMemoryBacked* memoryBacked) const
+			{
+				return ~0u;
+			}
```

Comment:
@Devsh-Graphics-Programming/seniors @Devsh-Graphics-Programming/mids this might interest you, you can constrain your Buffers to be created with ReBAR assumption (if the heap for a HOST_VISIBLE|DEVICE_LOCAL memory type is huge) and then it will skip using the staging buffer for uploads

Reply:
does not affect images as we have no host-image copy